File: app/src/main/java/com/xiaomai/geek/MainActivity.java
package com.xiaomai.geek;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.NavigationView;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.view.View;

import com.xiaomai.geek.ui.base.BaseActivity;
import com.xiaomai.geek.ui.module.password.PasswordListActivity;
import com.xiaomai.geek.ui.widget.MenuItemView;

public class MainActivity extends BaseActivity {

    private DrawerLayout mDrawerLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    protected int getLayoutResource() {
        return R.layout.activity_main;
    }

    @Override
    public void initViews() {
        super.initViews();

        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);

        initHeadViews();
    }

    private void initHeadViews() {
        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);

        View headerView = navigationView.getHeaderView(0);

        MenuItemView menuItemArticle = (MenuItemView) headerView.findViewById(R.id.menu_item_article);
        menuItemArticle.setSelected(true);
        menuItemArticle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });

        final MenuItemView menuItemPassword = (MenuItemView) headerView.findViewById(R.id.menu_item_password);
        menuItemPassword.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(mContext, PasswordListActivity.class));
            }
        });

        MenuItemView menuItemSettings = (MenuItemView) headerView.findViewById(R.id.menu_item_settings);
        menuItemSettings.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
    }

    public void openDrawer() {
        mDrawerLayout.openDrawer(GravityCompat.START);
    }
}

File: app/src/main/java/com/xiaomai/geek/main/MainActivity.kt
package com.xiaomai.geek.main

import android.os.Bundle
import android.support.v4.view.GravityCompat
import android.view.View
import com.xiaomai.geek.R
import com.xiaomai.geek.article.view.ArticleCategoryListFragment
import com.xiaomai.geek.base.BaseActivity
import com.xiaomai.geek.common.MenuItemView
import kotlinx.android.synthetic.main.main_activity.*

/**
 * Created by wangce on 2018/1/26.
 */
class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        title_view.setTitle(getString(R.string.article))

        title_view.setOnBackClickListener(View.OnClickListener {
            drawer_layout.openDrawer(GravityCompat.START)
        })

        val transaction = supportFragmentManager.beginTransaction()
        transaction.add(R.id.content, ArticleCategoryListFragment.newInstance(), ArticleCategoryListFragment::class.java.simpleName)
        transaction.commit()

        initDrawerView()
    }

    private fun initDrawerView() {
        val headerView = navigation_view.getHeaderView(0)
        val menuArticle = headerView.findViewById<MenuItemView>(R.id.menu_item_article)

        menuArticle.isSelected = true
        menuArticle.setOnClickListener {
            drawer_layout.closeDrawer(GravityCompat.START)
        }
    }

    override fun useBaseLayout(): Boolean {
        return false
    }

    override fun getLayoutId(): Int = R.layout.main_activity

    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
            return
        }
        super.onBackPressed()
    }
}